gms/gossiper.cc

```diff
@@ -1748,6 +1756,13 @@ future<> gossiper::start_gossiping(int generation_nbr, std::map<application_stat
     });
 }
 
+future<> gossiper::advertise_myself() {
+    if (_initial_generation_number.has_value()) {
```

Comment:
nit: you could drop the if and override unconditionally.

Reply:
Yes, we could drop it and later bring the check back because we need to do more stuffs inside the if branch.